packages/components/src/List/Toolbar/ColumnChooserButton/ColumnChooser/ColumnChooser.component.js

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { ColumnChooserProvider } from './columnChooser.context';
+import { useColumnChooserManager } from '../hooks';
+import FilterBar from '../../../../FilterBar';
+import ColumnChooserBody from './ColumnChooserBody';
+import ColumnChooserFooter from './ColumnChooserFooter';
+import ColumnChooserHeader from './ColumnChooserHeader';
+import getDefaultT from '../../../../translate';
+import cssModule from './ColumnChooser.scss';
+import { getTheme } from '../../../../theme';
+
+const theme = getTheme(cssModule);
+
+const haveColumnLabel = label => column => column.label.toLowerCase().includes(label.toLowerCase());
+const filterColumnsChooser = (columns, filter) => columns.filter(haveColumnLabel(filter));
+const changeVisibleToHidden = column => ({ ...column, hidden: !column.visible });
+const transformColumnsChooser = columns => columns.map(changeVisibleToHidden);
```

Comment:
```suggestion
const addVisibilityInColumns = columns => columns.map(changeVisibleToHidden);
```

Reply:
Hmmm the function transform the columns shape from the chooser to columns list shape. I see your point, I'll change it but not with your proposition.

Reply:
I see your point, add can give a false sense of mutability.
Ok for me if the function name is more explicit about what it does.